### ISSUE TITLE
Add components-configurations.sh

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/TF-PSA-Crypto#125]
Add component-configurations.sh in tests/scripts
Same components as in Mbed TLS but:
- component_build_baremetal (handled separately, see #117)
- component_test_no_platform, test_full_no_deprecated and test_full_no_deprecated_deprecated_warning  (handled separately, see #126) 
- build_mbedtls_config_file, not relevant to TF-PSA-Crypto and TF_PSA_CRYPTO(_USER)_CONFIG_FILE covered by a component in components-build-system.sh
- test_no_strings, not relevant to TF-PSA-Crypto

At time of writing: 12 components.
For `build_tfm` there is a dependency on https://github.com/Mbed-TLS/mbedtls/issues/9879.

Components based on make should be adapted to CMake. Compilation flags don't have to stay exactly the same but the ones relevant to the component should be kept the same.

In components:
- calls to `ssl-opt.sh`, `compat.sh`, `context-info.sh` and `programs/test/selftest` should be removed.
- out of scope for this issue, calls to `programs/test/cpp_dummy_build`, `run_demos.py`, `test_psa_constant_names.py` and `run-metatests.sh`.

**Comment by Harry-Ramsey** [comment]
I have managed to port some components over however TF-PSA-Crypto does not have a selftest.c file. Is this also to be removed?

**Comment by ronald-cron-arm** [comment]
> I have managed to port some components over however TF-PSA-Crypto does not have a selftest.c file. Is this also to be removed from components?

Yes, we do not need calls to programs/test/selftest in TF-PSA-Crypto. The self tests are anyway executed in test suites. But in the case of calloc maybe, see comment in test_malloc_0_null(), to be checked.

**Comment by ronald-cron-arm** [comment]
> Furthermore, are we still enabling the `build_tfm` component as TF-PSA-Crypto does not have a config for that either?

Yes, I've created https://github.com/Mbed-TLS/mbedtls/issues/9879 for the move of some crypto config files to TF-PSA-Crypto and updated the description of this issue with a mention to build_tfm and its dependency on https://github.com/Mbed-TLS/mbedtls/issues/9879.
